cmd/maya-apiserver/app/server/backup_endpoint_v1alpha1.go

```diff
@@ -0,0 +1,388 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/golang/glog"
+	"github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	"github.com/openebs/maya/pkg/client/generated/clientset/internalclientset"
+	snapshot "github.com/openebs/maya/pkg/snapshot/v1alpha1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type backupAPIOps struct {
+	req  *http.Request
+	resp http.ResponseWriter
+}
+
+// backupV1alpha1SpecificRequest deals with backup API requests
+func (s *HTTPServer) backupV1alpha1SpecificRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	backupOp := &backupAPIOps{
+		req:  req,
+		resp: resp,
+	}
+
+	switch req.Method {
+	case "POST":
+		return backupOp.create()
+	case "GET":
+		return backupOp.get()
+	}
+	return nil, CodedError(405, ErrInvalidMethod)
+}
+
+// Create is http handler which handles backup create request
+func (bOps *backupAPIOps) create() (interface{}, error) {
+	bkp := &v1alpha1.BackupCStor{}
+
+	err := decodeBody(bOps.req, bkp)
+	if err != nil {
+		return nil, err
+	}
+
+	// backup name is expected
+	if len(strings.TrimSpace(bkp.Spec.BackupName)) == 0 {
+		return nil, CodedError(400, fmt.Sprintf("Failed to create backup: missing backup name "))
+	}
+
+	// namespace is expected
+	if len(strings.TrimSpace(bkp.Namespace)) == 0 {
+		return nil, CodedError(400, fmt.Sprintf("Failed to create backup '%v': missing namespace", bkp.Spec.BackupName))
+	}
+
+	// volume name is expected
+	if len(strings.TrimSpace(bkp.Spec.VolumeName)) == 0 {
+		return nil, CodedError(400, fmt.Sprintf("Failed to create backup '%v': missing volume name", bkp.Spec.BackupName))
+	}
+
+	// backupIP is expected
+	if len(strings.TrimSpace(bkp.Spec.BackupDest)) == 0 {
+		return nil, CodedError(400, fmt.Sprintf("Failed to create backup '%v': missing backupIP", bkp.Spec.BackupName))
+	}
+
+	// snapshot name is expected
+	if len(strings.TrimSpace(bkp.Spec.SnapName)) == 0 {
+		return nil, CodedError(400, fmt.Sprintf("Failed to create backup '%v': missing snapName", bkp.Spec.BackupName))
+	}
+
+	openebsClient, _, err := loadClientFromServiceAccount()
+	if err != nil {
+		return nil, CodedError(400, fmt.Sprintf("Failed to create openEBSClient '%v'", err))
+	}
+
+	if err = createSnapshotForBackup(bkp); err != nil {
+		return nil, CodedError(400, fmt.Sprintf("Failed to create snapshot '%v'", err))
+	}
+
+	bkp.Name = bkp.Spec.SnapName + "-" + bkp.Spec.VolumeName
```

Comment:
backupcstor object should be created per iteration..

Reply:
ok.. got it.. snapname is different per iteration, so, backupcstor object name would be different